src/main/java/com/nashtech/exception/AppExceptionHandler.java

```diff
@@ -1,33 +1,37 @@
-package com.nashtech.vehicleapplication.exception;
+package com.nashtech.exception;
 
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.reactive.result.method.annotation.ResponseEntityExceptionHandler;
-
 import java.time.LocalDateTime;
 
 /**
  * Global exception handler for the application.
  */
 @ControllerAdvice
-public class AppExceptionHandler extends ResponseEntityExceptionHandler {
+public class AppExceptionHandler extends
+        ResponseEntityExceptionHandler {
 
     /**
-     * Exception handler for handling {@link WebClientCustomException}.
+     * Global exception handler for handling {@link DataNotFoundException}.
      *
-     * @param webClientCustomException
-     * The {@link WebClientCustomException} instance.
-     * @return A {@link ResponseEntity} containing the error response.
+     * @param dataNotFoundException
+     * The {@link DataNotFoundException} to be handled.
+     * @return A ResponseEntity
+     * containing an {@link ApiError} object with the error details.
      */
-    @ExceptionHandler(WebClientCustomException.class)
-    public ResponseEntity<Object> handleWebClientCustomException(
-            final WebClientCustomException webClientCustomException) {
-        return new ResponseEntity<>(new ApiError(webClientCustomException
-                .getMessage(),
+    @ExceptionHandler(DataNotFoundException.class)
+    public ResponseEntity<Object> handleDataNotFoundException(
+            final DataNotFoundException dataNotFoundException) {
```

Comment:
Include a check to see if "dataNotFoundException" object contains an exception message and if not provide a default message.

Reply:
Added the check to see if "dataNotFoundException" object contains an exception message, added the default message as well